packages/datadog-instrumentations/src/pg.js

```diff
@@ -31,7 +31,7 @@ function wrapQuery (query) {
     const asyncResource = new AsyncResource('bound-anonymous-fn')
     const processId = this.processID
     let pgQuery = {
-      text: arguments[0]
+      text: arguments[0] && typeof arguments[0] === 'object' ? arguments[0].text : arguments[0]
```

Comment:
What is the impact of this on https://github.com/DataDog/dd-trace-js/pull/3087/files#diff-d568b60c6ac1af79365d9cf6b40668f51f62b562a1841e2acd9b2b4af1cce1caR44? It seems that we would be overriding the object with the text property.

Reply:
@rochdev  This is to determine where we should be getting the query text from, the issue was with prepared statements the text query was being set to the entire object. This way we are checking if it is an object and then assign it based on that.

Reply:
But then on line 44 (linked above) the argument gets overridden. Won't that cause issues since the original method will now be getting a completely different first parameter?

Reply:
Should I be setting this a different way?

Reply:
I don't know this code well enough at this point, but we need to make sure we're not changing the structure of the `arguments` object which could have unintended side-effects. It's actually worrying that no tests are failing given that we're now overriding the first parameter.

Reply:
@rochdev here is a [new PR](https://github.com/DataDog/dd-trace-js/pull/3091) that addresses the above.